bindings/src/main/java/io/pravega/storage/filesystem/FileSystemStorageFactoryCreator.java

```diff
@@ -9,20 +9,44 @@
  */
 package io.pravega.storage.filesystem;
 
+import com.google.common.base.Preconditions;
 import io.pravega.segmentstore.storage.ConfigSetup;
 import io.pravega.segmentstore.storage.StorageFactory;
 import io.pravega.segmentstore.storage.StorageFactoryCreator;
+import io.pravega.segmentstore.storage.StorageFactoryInfo;
+import io.pravega.segmentstore.storage.StorageMetadataFormat;
+import io.pravega.segmentstore.storage.StorageLayoutType;
+
 import java.util.concurrent.ScheduledExecutorService;
 
 public class FileSystemStorageFactoryCreator implements StorageFactoryCreator {
 
     @Override
-    public String getName() {
-        return "FILESYSTEM";
+    public StorageFactoryInfo[] getStorageFactories() {
+        return new StorageFactoryInfo[]{
+                StorageFactoryInfo.builder()
+                        .name("FILESYSTEM")
+                        .storageMetadataFormat(StorageMetadataFormat.TABLE_BASED)
+                        .storageLayoutType(StorageLayoutType.CHUNKED_STORAGE)
+                        .build(),
+                StorageFactoryInfo.builder()
+                        .name("FILESYSTEM")
+                        .storageMetadataFormat(StorageMetadataFormat.HEADER_BASED)
+                        .storageLayoutType(StorageLayoutType.ROLLING_STORAGE)
+                        .build()
+        };
     }
 
     @Override
-    public StorageFactory createFactory(ConfigSetup setup, ScheduledExecutorService executor) {
-        return new FileSystemStorageFactory(setup.getConfig(FileSystemStorageConfig::builder), executor);
+    public StorageFactory createFactory(StorageFactoryInfo storageFactoryInfo, ConfigSetup setup, ScheduledExecutorService executor) {
```

Comment:
This piece of code seems the same as `ExtendedS3StorageFactoryCreator` and perhaps other places. Wondering if we can do this in a common method, and then pass the specific instance on each implementation instead of repeating the same logic everywhere.

Reply:
I think it would be ugly looking method which takes two Supplier lambdas that possibly take other lambdas etc .
This is unnecessarily complex. I already dislike the round about way things get initialized. 
 
I think the medicine is more harmful than illness itself in this case :)

``` java 
public StorageFactory createFactory(StorageFactoryInfo storageFactoryInfo, ConfigSetup setup, ScheduledExecutorService executor) {
      return commonFunction(()-> new ExtendedS3SimpleStorageFactory(setup.getConfig(ChunkedSegmentStorageConfig::builder),
                    setup.getConfig(ExtendedS3StorageConfig::builder),
                    executor),
                         ()-> return new ExtendedS3StorageFactory(setup.getConfig(ExtendedS3StorageConfig::builder), executor);
)
```